src/main/scala/com/databricks/spark/csv/DefaultSource.scala

```diff
@@ -113,36 +114,52 @@ class DefaultSource
 
     val charset = parameters.getOrElse("charset", TextFile.DEFAULT_CHARSET.name())
     // TODO validate charset?
-
     val inferSchema = parameters.getOrElse("inferSchema", "false")
-    val inferSchemaFlag = if(inferSchema == "false") {
+    val inferSchemaFlag = if (inferSchema == "false") {
       false
-    } else if(inferSchema == "true") {
+    } else if (inferSchema == "true") {
       true
     } else {
       throw new Exception("Infer schema flag can be true or false")
     }
 
+    val lineParsingOpts = LineParsingOpts(parameters)
+    val realNumParsingOpts = RealNumberParsingOpts(parameters)
+    val intNumParsingOpts = IntNumberParsingOpts(parameters)
+    val stringParsingOpts = StringParsingOpts(parameters)
+
+    val csvParsingOpts = if (!parameters.exists { case (k, v) =>
+      k.startsWith("csvParsingOpts.")
+    }) {
+      CSVParsingOpts(delimiter = delimiter,
+        quoteChar = quoteChar,
+        escapeChar = escapeChar,
+        ignoreLeadingWhitespace = ignoreLeadingWhiteSpaceFlag,
+        ignoreTrailingWhitespace = ignoreTrailingWhiteSpaceFlag)
+    } else {
+      CSVParsingOpts(parameters)
+    }
+
     CsvRelation(path,
-      headerFlag,
-      delimiter,
-      quoteChar,
-      escapeChar,
-      commentChar,
-      parseMode,
-      parserLib,
-      ignoreLeadingWhiteSpaceFlag,
-      ignoreTrailingWhiteSpaceFlag,
-      schema,
-      charset,
-      inferSchemaFlag)(sqlContext)
+      useHeader = headerFlag,
+      csvParsingOpts = csvParsingOpts,
+      lineExceptionPolicy = lineParsingOpts,
```

Comment:
should be the same. renamed it to indicate that this is a "per line" exception handling policy.